wzprof.go

```diff
@@ -10,30 +11,116 @@ import (
 	"unsafe"
 
 	"github.com/google/pprof/profile"
+	"github.com/tetratelabs/wazero"
 	"github.com/tetratelabs/wazero/api"
 	"github.com/tetratelabs/wazero/experimental"
 	"golang.org/x/exp/slices"
 )
 
+type stackIteratorMaker interface {
+	Make(mod api.Module, def api.FunctionDefinition, wasmsi experimental.StackIterator) experimental.StackIterator
+}
```

Comment:
Since this is used internally only, maybe just replace this abstraction with a function/closure?
```go
type Runtime struct {
    symbols           symbolizer
    makeStackIterator func(api.Module, api.FunctionDefinition, experimental.StackIterator) experimental.StackIterator
}
```